arcaflow_plugin_fio/fio_schema.py

```diff
@@ -56,48 +65,90 @@ def is_sync(self) -> bool:
+    r"(?:0x\d+)|"
+    r"(?:\d+(?:k|kb|ki|kib|m|mb|mi|mib|g|gb|gi|gib|t|tb|ti|tib|p|pb|pi|pib)*)"
+)
+size_pattern = re.compile(
+    rf"^{size_pattern_string}(?:,{size_pattern_string}){{0,2}}$", re.IGNORECASE
+)
+
+size_pattern_with_percent_string = (
+    rf"{size_pattern_string}|(?:[1-9][0-9]?%$|^100%)"
+)
+size_pattern_with_percent = re.compile(
+    rf"^{size_pattern_with_percent_string}$", re.IGNORECASE
+)
+
+size_range_pattern = re.compile(
+    rf"^(?:{size_pattern_string})-(?:{size_pattern_string})$",
+    re.IGNORECASE,
+)
+
+size_multi_range_pattern = re.compile(
+    rf"^(?:{size_pattern_string})-(?:{size_pattern_string})"
+    rf"(?:,(?:{size_pattern_string})-(?:{size_pattern_string})){{0,2}}$",
+    re.IGNORECASE,
+)
+
+
 @dataclass
 class JobParams:
+    # Units
+    kb_base: typing.Annotated[
+        typing.Optional[KbBase],
+        schema.name("Units Base"),
+        schema.description(
+            "Select the interpretation of unit prefixes in input parameters. The "
+            "default is 1024 (compatibility mode): Power-of-2 values with SI prefixes "
+            "(e.g., KB) and Power-of-10 values with IEC prefixes (e.g., KiB). When "
+            "setting this to 1000, inputs comply with IEC 80000-13 and the "
+            "International System of Units (SI): Power-of-2 values with IEC prefixes "
+            "(e.g., KiB) and Power-of-10 values with SI prefixes (e.g., KB)"
+        ),
+    ] = None
+
     # Job Description
     loops: typing.Annotated[
         typing.Optional[int],
         schema.name("Number of Job Loops"),
         schema.description(
             "Run the specified number of iterations of this job. Used to repeat the "
-            "same workload a given number of times."
+            "same workload a given number of times. Default is 1."
         ),
     ] = None
     numjobs: typing.Annotated[
         typing.Optional[int],
         schema.name("Number of Job Clones"),
         schema.description(
             "Create the specified number of clones of this job. Each clone of job is "
-            "spawned as an independent thread or process. May be used to setup a "
+            "spawned as an independent thread or process. May be used to set up a "
             "larger number of threads/processes doing the same thing. Each thread is "
-            "reported separately."
+            "reported separately. Default is 1."
         ),
     ] = None
 
     # Time related parameters
     runtime: typing.Annotated[
         typing.Optional[str],
+        validation.pattern(time_pattern),
         schema.name("Job Run Time"),
         schema.description(
             "Limit runtime. The test will run until it completes the configured I/O "
             "workload or until it has run for this specified amount of time, whichever "
             "occurs first. When the unit is omitted, the value is interpreted in "
-            "seconds."
+            "seconds. Default behavior is size-based operation."
```

Comment:
Rather than referring to "size-based", I would simply say that the default is an unlimited amount of time (or "not limited by time" or "no time limit", etc.).

Perhaps we don't need to specify a default behavior for this one.  (I.e., you're not currently describing the default for `rate_iops`...perhaps it's obvious what happens if one _doesn't_ specify a limit.)